tests/test_ci/TestControllers.py

```diff
@@ -487,3 +695,182 @@ def test_webhook_prerelease(self, mock_request):
                 data=json.dumps(data), headers=headers)
             last_release = CCExtractorVersion.query.order_by(CCExtractorVersion.released.desc()).first()
             self.assertNotEqual(last_release.version, '2.1')
+
+    @mock.patch('requests.get', side_effect=mock_api_request_github)
+    def test_webhook_push_no_after(self, mock_request):
+        """
+        Test webhook triggered with push event without 'after' in payload.
+        """
+        data = {'no_after': 'test'}
+        sig = generate_signature(str(json.dumps(data)).encode('utf-8'), g.github['ci_key'])
+        headers = generate_git_api_header('push', sig)
+        # one of ip address from github webhook
+        wsgi_environment = {'REMOTE_ADDR': '192.30.252.0'}
+        with self.app.test_client() as c:
+            response = c.post(
+                '/start-ci', environ_overrides=wsgi_environment,
+                data=json.dumps(data), headers=headers)
+
+    @mock.patch('requests.get', side_effect=mock_api_request_github)
+    @mock.patch('mod_ci.controllers.queue_test')
+    @mock.patch('mod_ci.controllers.GitHub')
+    @mock.patch('mod_ci.controllers.GeneralData')
+    def test_webhook_push_valid(self, mock_gd, mock_github, mock_queue_test, mock_request):
+        """
+        Test webhook triggered with push event without 'after' in payload.
```

Comment:
Copy paste ;)

Reply:
fixed in d068c47a924f02d8c93316c1d87d5e993ff7e325